src/charm.py

```diff
@@ -937,12 +949,13 @@ def is_blocked(self) -> bool:
     def _on_upgrade_charm(self, _) -> None:
         # Recreate k8s resources and add labels required for replication
         # when the pod loses them (like when it's deleted).
-        try:
-            self._create_services()
-        except ApiError:
-            logger.exception("failed to create k8s services")
-            self.unit.status = BlockedStatus("failed to create k8s services")
-            return
+        if self.upgrade.idle:
+            try:
+                self._create_services()
+            except ApiError:
+                logger.exception("failed to create k8s services")
```

Comment:
More of a question here, sorry :sweat_smile: but will `log.exception` (without the error as argument) display the error info for debugging purposes (eg. message, stack, etc)? I imagine so, just want to confirm so this doesn´t become a mistery later on!